src/CoreBundle/private/library/classes/TCMSTableEditor/TCMSTableWriter.class.php

```diff
@@ -127,19 +135,76 @@ protected function PostSaveHook(&$oFields, &$oPostTable)
     {
         parent::PostSaveHook($oFields, $oPostTable);
 
+        $newTableName = $oPostTable->sqlData['name'];
+
         // reset table list cache
         if (array_key_exists('_listObjCache', $_SESSION)) {
             $_SESSION['_listObjCache'] = array();
         }
 
         $this->getAutoclassesCacheWarmer()->updateTableById($this->sId);
 
-        // check table engine
-        $realEngine = $this->getRealTableEngine($oPostTable->sqlData['name']);
+        $this->adaptTableEngine($oPostTable, $newTableName);
+        $this->adaptTableName($this->sOldTblName, $newTableName);
+        $this->adaptTableComment($newTableName, $this->oldTableComment);
+    }
+
+    /**
+     * @param TCMSRecord $oPostTable
+     * @param string     $tableName
+     *
+     * @throws DBALException
+     */
+    private function adaptTableEngine(&$oPostTable, string $tableName): void
+    {
+        $realEngine = $this->getRealTableEngine($tableName);
         $requestedEngine = $this->getTableEngine($oPostTable->sqlData);
         if (strtolower($realEngine) !== strtolower($requestedEngine)) {
-            $this->changeTableEngine($oPostTable->sqlData['name'], $requestedEngine);
-            $this->SaveField('engine', $this->getRealTableEngine($oPostTable->sqlData['name']), false);
+            $this->changeTableEngine($tableName, $requestedEngine);
+            $newTableEngine = $this->getRealTableEngine($tableName);
+            $this->SaveField('engine', $newTableEngine, false);
+        }
+    }
+
+    /**
+     * @param string $oldTableName
+     * @param string $newTableName
+     *
+     * @throws DBALException
+     */
+    private function adaptTableName(string $oldTableName, string $newTableName): void
+    {
+        if ($oldTableName !== $newTableName) {
+            $databaseConnection = $this->getDatabaseConnection();
+
+            $query = sprintf('ALTER TABLE %s RENAME %s', $databaseConnection->quoteIdentifier($oldTableName), $databaseConnection->quoteIdentifier($newTableName));
+            $databaseConnection->executeQuery($query);
+
+            $aQuery = array(new LogChangeDataModel($query));
+            TCMSLogChange::WriteTransaction($aQuery);
+
+            $this->_RenameRelatedTables($this->sOldTblName, $newTableName);
```

Comment:
The old table name is both passed as argument and read from internal state - can we remove the argument?

Reply:
Also no change yet.